lang/src/org/partiql/lang/eval/EvaluatingCompiler.kt

```diff
@@ -974,13 +1021,16 @@ internal class EvaluatingCompiler(
                                 // Create a closure that groups all the rows in the FROM source into a single group.
                                 thunkFactory.thunkEnv(metas) { env ->
                                     // Evaluate the FROM clause
-                                    val fromProductions: Sequence<FromProduction> = sourceThunks(env)
+                                    var fromProductions: Sequence<FromProduction> = sourceThunks(env)
                                     val registers = createRegisterBank()
 
                                     // note: the group key can be anything here because we only ever have a single
                                     // group when aggregates are used without GROUP BY expression
                                     val syntheticGroup = Group(valueFactory.nullValue, registers)
 
+                                    if (limit != null) {
+                                        fromProductions = fromProductions.take(compileLimit(limit, env))
```

Comment:
Move `compileLimit` outside of thunk body.

Reply:
In the new commit, I separated out the compilation ([L955](https://github.com/partiql/partiql-lang-kotlin/blob/ad2aa3e73f549c3ff66605d254c2c5b7dfdbc277/lang/src/org/partiql/lang/eval/EvaluatingCompiler.kt#L955)) from the evaluation ([L892](https://github.com/partiql/partiql-lang-kotlin/blob/ad2aa3e73f549c3ff66605d254c2c5b7dfdbc277/lang/src/org/partiql/lang/eval/EvaluatingCompiler.kt#L892)) so the compilation isn't done from within the thunk evaluation.